common/rfb/pam.c

```diff
@@ -84,8 +87,22 @@ int do_pam_auth(const char *service, const char *username, const char *password)
   };
   pam_handle_t *h = 0;
   ret = pam_start(service, username, &conv, &h);
-  if (ret == PAM_SUCCESS)
+  if (ret == PAM_SUCCESS) {
+
+#ifdef PAM_XDISPLAY
+    /* Pass the display name to PAM modules. */
+    ret = pam_set_item(h, PAM_XDISPLAY, displayStr);
+    /* Note: PAM_XDISPLAY is only supported by modern versions of PAM */
+    if (ret== PAM_SUCCESS || ret == PAM_BAD_ITEM) {
+        ret = pam_authenticate(h, 0);
+    } else {
+        syslog(LOG_CRIT, "pam_set_item(PAM_XDISPLAY) failed: %d (%s)",
+               ret, pam_strerror(h, ret));
```

Comment:
We don't generally use syslog, so let's avoid making this some odd exception. Please use the normal logging.

Which might mean we need to merge `pam.c` in to `UnixPasswordValidator`. Not sure if there was something preventing that, and that's why we have separate files?

Reply:
Is probably because `libpam`, which is pure C code needs to have access to the functions. I will investigate this possibility.